clients/ethereum/client.go

```diff
@@ -363,6 +374,25 @@ func (c *client) ExecuteTransfer(
 	return txHash, err
 }
 
+// CheckClientAvailability will check the client availability and set the metric accordingly
+func (c *client) CheckClientAvailability(ctx context.Context) error {
+	currentBlock, err := c.clientWrapper.BlockNumber(ctx)
+	if err != nil {
+		c.clientWrapper.SetStringMetric(core.MetricEthereumClientStatus, ethElrond.Unavailable.String())
+		return err
+	}
+	if currentBlock == c.lastBlockNumber {
+		c.retriesAvailabilityCheck++
+		if c.retriesAvailabilityCheck >= c.allowDelta {
+			c.clientWrapper.SetStringMetric(core.MetricEthereumClientStatus, ethElrond.Unavailable.String())
```

Comment:
let's add a new metric that will output the last error encounter + will reset the error when everything is OK

Reply:
done